Gems/Atom/Feature/Common/Assets/Shaders/Reflections/ReflectionScreenSpaceFilter.azsl

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <scenesrg.srgi>
+#include <viewsrg.srgi>
+
+#include <Atom/Features/PostProcessing/FullscreenVertexUtil.azsli>
+#include <Atom/Features/PostProcessing/FullscreenVertexInfo.azsli>
+#include <Atom/Features/PostProcessing/FullscreenPixelInfo.azsli>
+#include <Atom/Features/MatrixUtility.azsli>
+#include <Atom/Features/PBR/LightingUtils.azsli>
+#include <Atom/Features/PBR/Microfacet/Fresnel.azsli>
+#include <Atom/Features/PostProcessing/PostProcessUtil.azsli>
+
+ShaderResourceGroup PassSrg : SRG_PerPass
+{
+    Texture2DMS<float>  m_depth;
+    Texture2DMS<float4> m_normal;       // RGB10 = EncodeNormalSignedOctahedron(worldNormal), A2 = multiScatterCompensationEnabled
+    Texture2DMS<float4> m_specularF0;   // RGB8 = SpecularF0, A8 = Roughness
+    Texture2D<float4>   m_traceReflection;
+    Texture2D<float4>   m_traceCoords;
+    Texture2D<float4>   m_history;
+
+    // output image scale and dimensions
+    float m_invOutputScale;
+    uint  m_outputWidth;
+    uint  m_outputHeight;
+    uint  m_mipLevels;
+
+    // SSR Settings
+    bool  m_coneTracing;
+    bool  m_rayTracing;
+    bool  m_temporalFiltering;
+    float m_invTemporalFilteringStrength;
+    float m_maxRoughness;
+    float m_roughnessBias;
+    bool  m_luminanceClamp;
+    float m_maxLuminance;
+
+    Sampler LinearSampler
+    {
+        MinFilter = Linear;
+        MagFilter = Linear;
+        MipFilter = Linear;
+        AddressU  = Clamp;
+        AddressV  = Clamp;
+        AddressW  = Clamp;
+    };
+
+    Sampler PointSampler
+    {
+        MinFilter = Point;
+        MagFilter = Point;
+        MipFilter = Point;
+        AddressU  = Clamp;
+        AddressV  = Clamp;
+        AddressW  = Clamp;
+    };
+}
+
+#include <Atom/RPI/ShaderResourceGroups/DefaultDrawSrg.azsli>
+#include "ReflectionCommon.azsli"
+
+// Vertex Shader
+VSOutput MainVS(VSInput input)
+{
+    VSOutput OUT;
+
+    float4 posTex = GetVertexPositionAndTexCoords(input.m_vertexID);
+    OUT.m_texCoord = float2(posTex.z, posTex.w);
+    OUT.m_position = float4(posTex.x, posTex.y, 0.0, 1.0);
+    return OUT;
+}
+
+// Pixel Shader
+PSOutput MainPS(VSOutput IN)
+{
+    float2 coords = IN.m_position.xy;
+    float2 UV = IN.m_texCoord.xy;
+    uint2  dimensions = uint2(PassSrg::m_outputWidth, PassSrg::m_outputHeight);
+    float2 fullScreenCoords = coords * PassSrg::m_invOutputScale;
+    uint2  fullScreenDimensions = dimensions * PassSrg::m_invOutputScale;
+
+    float4 result = float4(0.0f, 0.0f, 0.0f, 0.0f);
```

Comment:
Recommend adding a detailed description of what this pass/shader is trying to do at the top.